### ISSUE TITLE
Avoid pthread ”_np” functions for portability?

### ISSUE THREAD

**Issue by peter-jansson** [xrootd/xrootd#1645]
Compiling on Alpine Linux currently fails due to the non-existence of non-standard extensions of GNU to pthread functions. E.g., [`pthread_rwlockattr_setkind_np`](https://man7.org/linux/man-pages/man3/pthread_rwlockattr_setkind_np.3.html) is not in the musl C library. I recommmend to work around code like [this](https://github.com/xrootd/xrootd/blob/c3afc27d72c9b0460bfac05967d6a7ea8f9fa99a/src/XrdSys/XrdSysPthread.hh#L356), in XrdSysPthread, to welcome Alpine Linux users to the world of XRootD. Is it possible?

**Comment by simonmichal** [comment]
@peter-jansson : thanks for bringing this up! 

First of all, unfortunately I couldn't find any predefined macro that will tell us if we are using the musl C library (I am open to any suggestions). But let's say we do in cmake something like:
```
check_function_exists( pthread_rwlockattr_setkind_np, HAS_NP)
if( NOT HAS_NP )
  add_compile_definitions( __MUSL__ )
endif()
```
Then we can use the `__MUSL__` macro to make sure the `_np` functions are not used.

However, having this in place (and fixing few other small issues) I run into the following problems:
- `error: 'innetgr' was not declared in this scope`
- the `uuid/uuid.h` header is missing (AFAIU the there's not package providing it)
- `error: 'timeval' does not name a type`

Now, I'm not sure we can work around those problems, especially the missing `uuid/uuid.h` header (I couldn't find any package providing it).

If you are willing to contribute, please open a PR and we can try to address those issues one by one.

**Comment by abh3** [comment]
Where are we on this ticket?

**Comment by peter-jansson** [comment]
> * the `uuid/uuid.h` header is missing (AFAIU the there's not package providing it)

The [util-linux package](https://pkgs.alpinelinux.org/package/edge/main/x86_64/util-linux) on Alpine Linux contains uuid.h from [libuuid in the util-linux at kernel.org](https://git.kernel.org/pub/scm/utils/util-linux/util-linux.git/tree/libuuid/src).

**Comment by amadio** [comment]
Support for musl is (maybe partially) addressed in #1632, but I don't see any changes related to pthread functions there. We do intend to add support for it, although we cannot give a particular timeline, as we have to prioritize current supported platforms. I hope to test/merge musl support as part of the next feature release, 5.6.

**Comment by amadio** [comment]
Fixed with merge of musl branch into master (#1908). I also added a build on alpine to GitHub Actions.
